src/AppBundle/Controller/DonationController.php

```diff
@@ -21,8 +22,47 @@ class DonationController extends Controller
      */
     public function indexAction(Request $request)
     {
-        return $this->render('donation/index.html.twig', [
-            'amount' => (float) $request->query->get('montant', 50),
+        if (!$amount = $request->query->get('montant')) {
+            return $this->render('donation/index.html.twig', [
+                'amount' => (float) $request->query->get('montant', 50),
+            ]);
+        }
+
+        if ($isMonthlyPayment = $request->get('abonnement', false)) {
```

Comment:
Usage of the `get()` method is dangerous as we don't see if `abonnement` comes from the path variable, or query string or post parameters. You should use the right request parameter bag instead.